pkg/engine/jsonContext.go

```diff
@@ -26,7 +29,7 @@ func LoadContext(logger logr.Logger, contextEntries []kyverno.ContextEntry, resC
 	policyName := ctx.Policy.Name
 	if store.GetMock() {
 		rule := store.GetPolicyRuleFromContext(policyName, ruleName)
-		if len(rule.Values) == 0 {
+		if rule == nil || len(rule.Values) == 0 {
```

Comment:
Noticed this bug here. `store.GetPolicyRuleFromContext(policyName, ruleName)` above may return a nil pointer which causes rule.Values to panic if not defined during `kyverno test`. We should also skip context var substitution if the rule is left entirely undefined in the variables.yaml.